src/tests/collab.test.tsx

```diff
@@ -47,26 +42,24 @@ jest.mock("socket.io-client", () => {
 
 describe("collaboration", () => {
   it("creating room should reset deleted elements", async () => {
-    importFromLocalStorageSpy.mockImplementation(() => ({
-      elements: [
-        API.createElement({ type: "rectangle", id: "A" }),
-        API.createElement({
-          type: "rectangle",
-          id: "B",
-          isDeleted: true,
-        }),
-      ],
-      appState: null,
-    }));
-    await render(<AppWithCollab />);
+    Storage.prototype.getItem = jest.fn((key) => {
+      let res = null;
+      if (key === STORAGE_KEYS.LOCAL_STORAGE_ELEMENTS) {
+        res = [
+          API.createElement({ type: "rectangle", id: "A" }),
+          API.createElement({
+            type: "rectangle",
+            id: "B",
+            isDeleted: true,
+          }),
+        ];
+      }
 
-    await waitFor(() => {
-      expect(h.elements).toEqual([
-        expect.objectContaining({ id: "A" }),
-        expect.objectContaining({ id: "B", isDeleted: true }),
-      ]);
-      expect(API.getStateHistory().length).toBe(1);
```

Comment:
I'd keep this to ensure we don't regress

Reply:
since we are mocking LS now so this assertion will fail as we call `clearElementsForLocalStorage` when importing from LS

Reply:
Right, well that's a problem as it's integral to this test (and the app behavior itself).